src/auth/AuthRegistry.cc

```diff
@@ -163,7 +166,13 @@ void AuthRegistry::get_supported_methods(
       *methods = client_methods;
     }
     if (modes) {
-      *modes = client_modes;
+      switch (peer_type) {
+      case CEPH_ENTITY_TYPE_MON:
```

Comment:
Shouldn't this be `CEPH_ENTITY_TYPE_CLIENT`?

Reply:
No, the outer bit is who we are (cct->get_module_type()), and this is who we are connecting to (mon).

Reply:
ah, got it.